src/libs/blueprint/conduit_blueprint_mesh_topology_metadata.hpp

```diff
@@ -0,0 +1,613 @@
+ G(1,0) = {0,3}{3,4}{4,1}{1,0}{1,10}{10,9}{9,0}{4,13}{13,10}...
+                     |         |                |     |
+                     |         |                |     |
+                     ----------------------------------
+                     |
+                     edges that are in face 2. 
+
+
+ What does G(e,a) mean?
+
+                               Association (a)
+  ---------------------------------------------------------------------------
+     | 3              | 2               | 1               | 0
+  ---------------------------------------------------------------------------
+E  3 |                | face ids in     | edge ids for    | point ids in
+n    |    self        | element i       | element i       | element i
+t    |                | BBC*            |                 | dim[3] connectivity
+i  --|-----------------------------------------------------------------------
+t  2 | element ids    |                 | edge ids in     | point ids in 
+y    | that contain   |      self       | face i          | face i
+     | face i         |                 | BBC *           | dim[2] connectivity
+e  --|-----------------------------------------------------------------------
+   1 | element ids    | face ids that   |                 | point ids in
+     | that contain   | contain edge i  |      self       | edge i
+     | edge i         |                 |                 | dim[1] connectivity
+   --|-----------------------------------------------------------------------
+   0 | element ids    | face ids that   | edge ids that   |
+     | that contain   | contain point i | contain point i |     self
+     | point i        |                 |                 |
+  ---------------------------------------------------------------------------
+
+ *BBC = built by cascade
+
+ There is another mapping concept called LOCAL mapping L(e,a) where entity
+ gets its own unique index ordering. The local maps look essentially like an
+ increasing series of integers, though the number of repetitions for an
+ entity depends on e,a.
+
+ Example:
+ L(3,2) = {0,1,2,3,4,5},{6,7,8,9,10,11},...
+
+               6 faces 
+           ----------------
+           |              |
+ L(2,3) = {0}{0}{0}{0}{0}{0}{1}{1}{1}{1}{1}{1}...
+
+ */
+class CONDUIT_BLUEPRINT_API TopologyMetadata : public TopologyMetadataBase
+{
+public:
+    //-----------------------------------------------------------------------
+    /**
+     @brief Legacy constructor, which builds all of the topology levels in the shape
+            cascade as well as all associations.
+
+     @param topology     The input topology node.
+     @param coordset     The input coordset associated with the topology.
+     */
+    TopologyMetadata(const conduit::Node &topology, const conduit::Node &coordset);
+
+    //-----------------------------------------------------------------------
+    /**
+     @brief Constructor for the NewTopologyMetadata class. This constructor
```

Comment:
```suggestion
     @brief Constructor for the TopologyMetadata class. This constructor
```
If you write `NewTopologyMetadata`, I would be looking for the class to be called that.

Reply:
Oops. I originally wrote this class outside Conduit with that name. I'll fix it.